crates/typst-library/src/foundations/symbol.rs

```diff
@@ -244,9 +245,57 @@ impl Debug for List {
     }
 }
 
+fn repr_variants<'a, 'b>(
+    variants: impl Iterator<Item = (&'a str, char)>,
+    applied_modifiers: impl Iterator<Item = &'b str> + Clone,
+) -> String {
+    crate::foundations::repr::pretty_array_like(
+        &variants
+            .filter(|(variant, _)| {
+                // Only keep variants that can still be accessed, i.e., variants
+                // that contain all applied modifiers.
+                applied_modifiers
+                    .clone()
+                    .all(|am| variant.split('.').any(|m| m == am))
+            })
+            .map(|(variant, c)| {
+                let trimmed_variant = variant
+                    .split('.')
+                    .filter(|&m| applied_modifiers.clone().all(|am| m != am))
+                    .collect::<Vec<_>>();
```

Comment:
This logic seems like it would work well for the `envelope` example specifically, but for other examples it would probably break since it's not considering order, so something like `symbolname.variantA.variantB` and `symbolname.variantB.variantA` would be treated equally when trimming applied modifiers. The logic should have some zipping / common prefix behavior instead. We should also have a test for longer symbols where variants with the same name but different order are present.

Reply:
I don't think order matters when selecting the variant.
https://github.com/typst/typst/blob/bb0c8140950b3eec020a4f0147bbc4ea65f3952a/crates/typst-library/src/foundations/symbol.rs#L322-L335

Reply:
```typ
#let s = symbol(
  ("foo.bar", "A"),
  ("bar.foo", "B"),
)

#s.foo
#s.bar
#s.foo.bar
#s.bar.foo
```
renders as "A A A A"

Reply:
> ```typst
> #let s = symbol(
>   ("foo.bar", "A"),
>   ("bar.foo", "B"),
> )
> 
> #s.foo
> #s.bar
> #s.foo.bar
> #s.bar.foo
> ```
> 
> renders as "A A A A"

Uhhh. Is this not a bug?

Reply:
Modifiers are conceptually a set, not a list, so the symbol constructor call is bad here. It should probably error.

Reply:
Hmmm... Interesting. If this means that "foo.bar" would always be able to be called as "bar.foo", then, to be honest, I find that quite confusing, as the order itself can mean something. But if not and this only matters internally, then I'd be fine with it (albeit a bit confused still).

Reply:
You can always call `s.foo.bar` as `s.bar.foo`. We take care during symbol naming to not introduce any order dependence. The idea is that you don't have to care about the order, you care about properties of the symbol. Like, "I want a symbol that is an arrow, that is dashed, and points to the top." So you write `arrow.` (see the autocomplete), add `.dashed` (see the autocomplete), add `.t`. And then you have `arrow.dashed.t` and you don't care that the definition is actually `arrow.t.dashed`. The autocomplete will always show you all currently available modifiers rather than the list of all symbols, so you can explore modifier by modifier, without any regard for the order. It's really a set more than a list.